### ISSUE TITLE
SSHConnection should subclass asyncio.Protocol

### ISSUE THREAD

**Issue by danielzgtg** [ronf/asyncssh#262]
`asyncssh.connection.SSHConnection` implements the methods of `asyncio.Protocol` but doesn't inherit from it. This makes type checkers complain.

# Actual behavior

Consider the following sample code:

```python
import asyncio
import asyncssh

def custom_connect_logic(protocol: asyncio.Protocol):
    """Logic in the networking layer of my application"""

conn = asyncssh.SSHClientConnection(
    host='127.0.0.1', port=22,
    loop=asyncio.get_event_loop(),
    options=asyncssh.SSHClientConnectionOptions(),
)

custom_connect_logic(conn)
```

`conn` on the last line is highlighted yellow with a warning in PyCharm.

# Expected behavior

It would be nice to have this warning go away.

# Environment

## requirements.txt

```
asyncssh==2.1.0
```

## `python --version`

```
Python 3.8.0
```

## `lsb_release -a`

```
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu Focal Fossa (development branch)
Release:        20.04
Codename:       focal
```

**Comment by ronf** [comment]
Thanks for the suggestion. I did a quick test and it looks like making SSHConnection explicitly inherit from asyncio.Protocol doesn't cause any problems and I agree that it could be helpful for type checkers. I'm in the middle of some other changes right now, but I'll get this change into the "develop" branch as soon as the other work is complete and it'll be rolled into the next release.

**Comment by ronf** [comment]
Ok - this change is commit cd7248a in the "develop" branch.

When I actually made the change, it reminded me why I didn't do it before. The signature of data_received() in SSHConnection is actually different from the one in asyncio.Protocol. It takes an optional "datatype" argument, to allow tunneled SSH connections. Due to this extra argument, the pylint arguments-differ warning had be suppressed on SSHConnection.data_received(). However, to make type checkers happy and avoid the pylint warning, I've gone ahead with that.

**Comment by ronf** [comment]
This change is now available in AsyncSSH 2.2.0.
